### PR TITLE
Use Codecov for code coverage reporting

### PR DESCRIPTION
**Pull request by ofek**
### What does this PR do?

This service provides a rich UI for displaying code coverage at various resolutions: project-wide, per-directory/package, user-defined flags (e.g. unit, integration, docker), line-by-line, etc. You can also control CI & PR pass/fail based on coverage percentage. There is also the ability to use a pull request comment bot and dictate its contents. And there is a Chrome extension that overlays stats on GitHub's UI.

Also, AppVeyor now reports coverage too so you get a fuller picture. Codecov automatically merges reports.

### Motivation

Gain better insight into health of repo & encourage mentality of testing. Here is what `integrations-core` looks like: https://codecov.io/github/DataDog/integrations-core?branch=master

### Note

I got the idea that this will benefit the Agent while reading about how Elastic does coverage for all of their Golang stuff https://www.elastic.co/blog/code-coverage-for-your-golang-system-tests

### CODE REVIEW

**Comment by ofek** [comment]
^ there's the bot. Here is what the UI looks like 😄 https://codecov.io/gh/DataDog/datadog-agent/tree/a08db85b7c941e81c828ab14fae7231ec63c78d3/pkg

Currently, only unit tests are measured since that's the only stage that uses `--coverage` and I didn't want to make any changes.

**Comment by xvello** [comment]
Great idea, thanks!

File: appveyor.yml [hunk 1]
```diff
@@ -38,4 +39,5 @@ build: off
 test_script:
   - cd %APPVEYOR_BUILD_FOLDER%
   - inv -e deps
-  - inv -e test --race --profile --fail-on-fmt
+  - inv -e test --coverage --race --profile --fail-on-fmt
+  - codecov -f profile.cov
```
Comment (xvello) [inline appveyor.yml]: Is there a way to also make sure we don't fail the pipeline on this?

Comment (ofek) [reply appveyor.yml]: The command never fails (though there is a flag for that).

**Review by xvello — APPROVED** [review]
LGTM  👍

**Review by derekwbrown — APPROVED** [review]

**Pull request merged**